base_layer/chat_ffi/src/messages.rs

```diff
@@ -147,15 +147,16 @@ pub unsafe extern "C" fn message_vector_get_at(
     }
 
     let messages = &(*messages);
+    let position = position as usize;
```

Comment:
If `position` is set to usize::MAX, `position + 1` will either panic or overflow to 0, in which case `messages.0[usize::MAX]` will probably panic.

```rust
if messages.0.is_empty() {
// error
}
let len = messages.0.len() - 1;
if position as usize > len {
// error
}
```

`position as usize` should be ok as it _should_ be defined as a u32 with some rare exceptions according to the `c_uint` docs. To be safer you could use `usize::try_from`